bin/bdii-update

```diff
@@ -271,411 +283,439 @@ def read_ldif(source):
         sys.exit(0)
 
+
 def get_dns(ldif):
     dns = {}
     last_dn_index = len(ldif)
-    while ( 1 ):
-        dn_index = ldif.rfind("dn:",0,last_dn_index)
-        if ( dn_index == -1):
+    while 1:
+        dn_index = ldif.rfind("dn:", 0, last_dn_index)
+        if dn_index == -1:
             break
-        end_dn_index = ldif.find("\n",dn_index, last_dn_index) 
-        dn = ldif[dn_index + 4 :end_dn_index].lower()
-        dn = re.sub("\s*,\s*",",",dn)
-        dn = re.sub("\s*=\s*","=",dn)
-        dn = dn.replace("\\5c","\\\\") # Replace encoded slash
-        dn = dn.replace("\\2c","\\,") # Replace encoded comma
-        dn = dn.replace("\\3d","\\=") # Replace encoded equals
-        dn = dn.replace("\\2b","\\+") # Replace encoded plus
-        dn = dn.replace("\\3b","\\;") # Replace encoded semi colon
-        dn = dn.replace("\\22","\\\"") # Replace encoded quote
-        dn = dn.replace("\\3e","\\>") # Replace encoded greater than
-        dn = dn.replace("\\3c","\\<") # Replace encoded less than
-        end_entry_index = ldif.find("\n\n",dn_index, last_dn_index) 
+        end_dn_index = ldif.find("\n", dn_index, last_dn_index)
+        dn = ldif[dn_index + 4:end_dn_index].lower()
+        dn = re.sub("\s*,\s*", ",", dn)
+        dn = re.sub("\s*=\s*", "=", dn)
+        # Replace encoded slash
+        dn = dn.replace("\\5c", "\\\\")
+        # Replace encoded comma
+        dn = dn.replace("\\2c", "\\,")
+        # Replace encoded equals
+        dn = dn.replace("\\3d", "\\=")
+        # Replace encoded plus
+        dn = dn.replace("\\2b", "\\+")
+        # Replace encoded semi colon
+        dn = dn.replace("\\3b", "\\;")
+        # Replace encoded quote
+        dn = dn.replace("\\22", "\\\"")
+        # Replace encoded greater than
+        dn = dn.replace("\\3e", "\\>")
+        # Replace encoded less than
+        dn = dn.replace("\\3c", "\\<")
         dns[dn] = (dn_index, last_dn_index, end_dn_index)
         last_dn_index = dn_index
     return dns
 
+
 def group_dns(dns):
     grouped = {}
     for dn in dns:
         index = dn.rfind(",")
-        root = dn[index +1 :].strip()
-        if grouped.has_key(root):
+        root = dn[index + 1:].strip()
+        if root in grouped:
             grouped[root].append(dn)
         else:
             if root in config['BDII_PASSWD']:
-                grouped[root] = [ dn ]
+                grouped[root] = [dn]
             else:
                 if "o=shadow" in config['BDII_PASSWD'] and root == "o=grid":
-                    grouped[root] = [ dn ]
+                    grouped[root] = [dn]
                 elif root != "o=shadow":
-                    log.error("dn suffix %s in not specified in the slapd configuration file." % (root))
-
+                    log.error(("dn suffix %s in not specified in the slapd"
+                               " configuration file.") % root)
 
     return grouped
 
+
 def convert_entry(entry_string):
+    # multivalued = ['objectclass',
```

Comment:
seems so long and was probably quite painful so I was thinking there could be a need for it somewhere :)
Will drop it you are right, it will stay in history if needed